app/src/main/java/org/simple/clinic/drugs/selection/custom/drugfrequency/SelectDrugFrequencyDialog.kt

```diff
@@ -0,0 +1,91 @@
+package org.simple.clinic.drugs.selection.custom.drugfrequency
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.R
+import org.simple.clinic.di.injector
+import org.simple.clinic.drugs.search.DrugFrequency
+import org.simple.clinic.drugs.search.DrugFrequency.BD
+import org.simple.clinic.drugs.search.DrugFrequency.OD
+import org.simple.clinic.drugs.search.DrugFrequency.QDS
+import org.simple.clinic.drugs.search.DrugFrequency.TDS
+import org.simple.clinic.drugs.search.DrugFrequency.Unknown
+import org.simple.clinic.navigation.v2.Router
+import org.simple.clinic.navigation.v2.ScreenKey
+import org.simple.clinic.navigation.v2.Succeeded
+import org.simple.clinic.util.unsafeLazy
+import javax.inject.Inject
+
+class SelectDrugFrequencyDialog : DialogFragment() {
+  @Inject
+  lateinit var router: Router
+
+  private val screenKey: Key by unsafeLazy { ScreenKey.key(this) }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    var selectedValueIndex = frequenciesArrayIndexFromDrugFrequency(screenKey.drugFrequency)
```

Comment:
We are tracking a state as a mutable property here. Also, if there is a state we should switch to using Mobius loop. We go with a standard dialog fragment if there is no state and it's just handling a normal dialog.

Reply:
Also, is there a reason for us to do this in the first place 🤔 ? The dialog internally takes care of maintaining selection states. So we just need to get the selected frequency when the "Done" button is clicked. So, we can probably remove this tracking altogether.

Reply:
Based on whether or not we need to track the state, please change the approach.